# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../eglibc/x86-i686.patch
# Copyright (C) 2004 - 2012 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

__i686 is used in assembler, but when optimizing for a 686-class CPU
gcc defines __i686 as cpp macro ...

  - Rene Rebe <rene@exactcode.de>

--- libc/csu/defs.awk.orig	2000-01-01 00:00:00.000000000 +0000
+++ libc/csu/defs.awk	2012-12-24 21:23:06.612034400 +0000
@@ -24,4 +24,5 @@
 
     print "#include <libc-symbols.h>";
     print "weak_extern (__gmon_start__)";
+    print "#undef __i686";
 }

--- libc/sysdeps/i386/bp-asm.h.orig	2000-01-01 00:00:00.000000000 +0000
+++ libc/sysdeps/i386/bp-asm.h	2012-12-24 21:23:06.612034400 +0000
@@ -22,6 +22,8 @@
 #ifndef _bp_asm_h_
 # define _bp_asm_h_ 1
 
+#undef __i686
+
 # if __ASSEMBLER__
 
 #  if __BOUNDED_POINTERS__
